### PR TITLE
[GH1] Add sparse related changes to merge rules

### PR DESCRIPTION
**Pull request by janeyx99**
This PR would allow Quansight sparse experts (in addition to metamates) to approve sparse related changes. As the sparse module is relatively new and should not have many internal dependencies, we can start encouraging more GitHub 1st (GH1) landing for these. 

This is DIFFERENT from the superuser rule because it allows non-metamates to be approvers.

### CODE REVIEW

File: .github/merge_rules.json [hunk 1]
```diff
@@ -41,6 +41,27 @@
...
+         "torch/**/sparse/**",
+         "torch/**/*sparse*",
+         "torch/include/**/*Sparse*",
+         "torch/include/**/*sparse*",
+         "torch/optim/sparse*",
+         "torch/ao/nn/sparse/**",
+         "torch/utils/benchmark/**/*sparse*",
+         "aten/src/ATen/native/ao_sparse/**",
+         "aten/src/ATen/native/sparse/**",
+         "aten/src/ATen/**/*Sparse*",
```
Comment (pearu) [inline .github/merge_rules.json:59]: I suggest adding the following patterns if ok with @cpuhrsch :
```suggestion
         "aten/src/ATen/**/*Sparse*",
         "aten/src/ATen/*Sparse*",
         "torch/_masked/**",
         "test/*_masked*",
```

Comment (cpuhrsch) [reply .github/merge_rules.json:59]: Ok with me

File: .github/merge_rules.json [hunk 2]
```diff
@@ -41,6 +41,27 @@
...
+         "test/**/*sparse*"
+      ],
+      "approved_by": ["nikitaved", "cpuhrsch", "pearu"],
```
Comment (cpuhrsch) [inline .github/merge_rules.json:62]: Can we also add IvanYashchuk here?

File: .github/merge_rules.json [hunk 3]
```diff
@@ -41,6 +41,28 @@
...
+         "torch/optim/sparse*",
+         "torch/ao/nn/sparse/**",
+         "torch/utils/benchmark/**/*sparse*",
+         "aten/src/ATen/native/ao_sparse/**",
+         "aten/src/ATen/native/sparse/**",
+         "aten/src/ATen/**/*Sparse*",
+         "aten/src/ATen/*Sparse*",
+         "torch/_masked/**",
+         "test/*_masked*",
+         "test/**/*sparse*"
```
Comment (IvanYashchuk) [inline .github/merge_rules.json:61]: Covering `test/test_sparse.py` and `test/test_sparse_csr.py` would be helpful.
```suggestion
         "test/**/*sparse*"
         "test/*sparse*"
```

**Review by IvanYashchuk — APPROVED** [review]
Thank you, Jane!

File: .github/merge_rules.json (hunk 3 above)
Comment (janeyx99) [reply .github/merge_rules.json:61]: I think `test/**/*sparse*` already covers `test/*sparse*`

**Review by osalpekar — APPROVED** [review]

**Pull request closed without merging**